swim/node.go

```diff
@@ -517,14 +519,23 @@ func (n *Node) pingNextMember() {
 	n.logger.WithField("target", target).Info("ping request target reachable")
 }
 
-// GetReachableMembers returns a slice of members currently in this node's
-// membership list that aren't faulty.
-func (n *Node) GetReachableMembers() []string {
-	return n.memberlist.GetReachableMembers()
+// GetReachableMembers returns a slice of members containing only the reachable
+// members that satisfies the predicates passed in.
+func (n *Node) GetReachableMembers(predicates ...MemberPredicate) []Member {
+	predicates = append(predicates, reachableMember)
```

Comment:
That is true, but I don't think there is a prepend function to an array so I went for this one.

An other thing that I have been thinking is making a combiner predicate that takes a predicate and a variadic predicates and tests them all. This would also 'prevent' list expansion/memory allocation. But for both I think we need some more measurements and testing.

Since in a stable ring there should be only nodes in alive and suspect state I think the optimization of moving it to the front is needless at this point till we run into a case where it will be actually make a difference, but that needs a good benchmarking suite.